### PR TITLE
Introduced QueryBuilder approach

### PR DESCRIPTION
**Pull request by xSAVIKx**
refactored Query and Client classes, so that now:
API key, base URL and search path are now related to Client.
Query object is now just holder for search pattern and related parameters (start and end time, document language)
Introduced QueryBuilder that follows Builder pattern and should be used for Query object creation.
refactored all related to changes unit tests
rewritten examples to use new approach
refactored README to compile successfully in IDEA and to follow changes

### CODE REVIEW

**Comment by twingly-mob** [comment]
Slightly off-topic question, we are curious on https://github.com/twingly/twingly-search-api-java/pull/14/files#diff-04c6e90faac2675aa89e2176d2eec7d8R56 what does it do? It seems like it builds a jar file, but we can't find the key embedded in it. Where should we find the key?

File: src/test/groovy/com/twingly/search/QueryBuilderSpockTest.groovy [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import com.twingly.search.domain.Language
+import com.twingly.search.exception.QueryException
+import spock.lang.Specification
+
+import java.text.SimpleDateFormat
+
+class QueryBuilderSpockTest extends Specification {
+    private final SimpleDateFormat simpleDateFormat = new SimpleDateFormat(Constants.DATE_FORMAT);
+
+    def "should through QueryException for empty search pattern"() {
```
Comment (twingly-mob) [inline src/test/groovy/com/twingly/search/QueryBuilderSpockTest.groovy]: through -> throw

File: src/test/groovy/com/twingly/search/QueryBuilderSpockTest.groovy [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    def "should set documentLanguage from Language enum"() {
+        given:
+        def language = Language.English
+        when:
+        def query = QueryBuilder.create("pattern").documentLanguage(language).build()
+        then:
+        query.documentLanguage == Language.English.isoCode
+    }
+
+    def "should through QueryException on invalid search pattern set"() {
```
Comment (twingly-mob) [inline src/test/groovy/com/twingly/search/QueryBuilderSpockTest.groovy]: should be throw here as well

File: src/test/groovy/com/twingly/search/QueryBuilderSpockTest.groovy [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    def "should not set empty language"() {
+        when:
+        def query = QueryBuilder.create("pattern").documentLanguage("").build()
+        then:
+        query.documentLanguage == null
+    }
+
+
```
Comment (twingly-mob) [inline src/test/groovy/com/twingly/search/QueryBuilderSpockTest.groovy]: one newline too much

**Comment by xSAVIKx** [comment]
> Slightly off-topic question, we are curious on https://github.com/twingly/twingly-search-api-java/pull/14/files#diff-04c6e90faac2675aa89e2176d2eec7d8R56 what does it do? It seems like it builds a jar file, but we can't find the key embedded in it. Where should we find the key?

It's just an example how key should be passed to any jvm-based program.
Maybe, it'll be more clear, if the example will be smth like:
`java -jar myJar.jar -DTWINGLY_SEARCH_KEY=some_key_value`

**Comment by twingly-mob** [comment]
> It's just an example how key should be passed to any jvm-based program.
> Maybe, it'll be more clear, if the example will be smth like:
> java -jar myJar.jar -DTWINGLY_SEARCH_KEY=some_key_value

Yes, that's clearer.

**Comment by xSAVIKx** [comment]
@twingly-mob I've updated documentation with new example.

File: src/test/groovy/com/twingly/search/client/UrlConnectionClientSpockTest.groovy [hunk 4]
```diff
@@ -33,6 +36,41 @@ class UrlConnectionClientSpockTest extends Specification {
...
+    }
+
+    def "should create client with API key from System properties"() {
+        given:
+        System.setProperty(Constants.TWINGLY_API_KEY_PROPERTY, API_KEY)
+        when:
+        new UrlConnectionClient()
+        then:
+        notThrown(TwinglySearchException)
+        System.properties.remove(Constants.TWINGLY_API_KEY_PROPERTY)
```
Comment (twingly-mob) [inline src/test/groovy/com/twingly/search/client/UrlConnectionClientSpockTest.groovy]: WDYT of moving the removal of the property to `cleanup:`?

File: src/test/groovy/com/twingly/search/client/UrlConnectionClientSpockTest.groovy [hunk 5]
```diff
@@ -245,7 +281,7 @@ class UrlConnectionClientSpockTest extends Specification {
 
     def "should process query and return result isntance"() {
```
Comment (twingly-mob) [inline src/test/groovy/com/twingly/search/client/UrlConnectionClientSpockTest.groovy]: isntance -> instance

**Comment by dentarg** [comment]
@xSAVIKx great work, we will take a last quick look tomorrow, I think we will be happy :)

**Pull request merged**